src/main/java/hudson/plugins/build_timeout/BuildTimeoutWrapper.java

```diff
@@ -193,6 +195,8 @@ public synchronized void rescheduleIfScheduled() {
             public synchronized boolean tearDown(AbstractBuild build, BuildListener listener) throws IOException, InterruptedException {
                 if (task != null) {
                     task.cancel();
+                    // avoid memory leaks for the case where this timer is in the future (JENKINS-31627).
+                    Trigger.timer.purge();
```

Comment:
I don't get the connection between `Trigger.timer.purge()` and the `task` being canceled. Is `Trigger.timer` a `ThreadLocal` that has some relevant context that allows it to purge the correct `TimerTask`?